libenkf/src/enkf_state_nodes.c

```diff
@@ -0,0 +1,377 @@
+*/
+
+static void enkf_state_internalize_eclipse_state(enkf_state_type * enkf_state ,
+						 forward_load_context_type * load_context,
+						 const model_config_type * model_config ,
+                                                 int report_step ,
+                                                 bool store_vectors) {
+
+  shared_info_type   * shared_info   = enkf_state->shared_info;
+  const ecl_config_type * ecl_config = shared_info->ecl_config;
+  const run_arg_type * run_arg = forward_load_context_get_run_arg( load_context );
+  enkf_fs_type * result_fs = run_arg_get_result_fs( run_arg );
+  if (ecl_config_active( ecl_config ) && model_config_internalize_state( model_config , report_step )) {
+    member_config_type * my_config     = enkf_state->my_config;
+    const int  iens                    = member_config_get_iens( my_config );
+    const bool internalize_state       = model_config_internalize_state( model_config , report_step );
+
+    forward_load_context_load_restart_file( load_context , report_step);
+
+    /******************************************************************/
+    /**
+        Starting on the enkf_node_forward_load() function calls. This
+        is where the actual loading is done. Observe that this loading
+        might involve other load functions than the ones used for
+        loading PRESSURE++ from ECLIPSE restart files (e.g. for
+        loading seismic results..)
+    */
+
+    {
+      hash_iter_type * iter = hash_iter_alloc(enkf_state->node_hash);
+      while ( !hash_iter_is_complete(iter) ) {
+        enkf_node_type * enkf_node = hash_iter_get_next_value(iter);
+        if (enkf_node_get_var_type(enkf_node) == DYNAMIC_STATE &&
+            enkf_node_get_impl_type(enkf_node) == FIELD) {
+
+          bool internalize_kw = internalize_state;
+          if (!internalize_kw)
+            internalize_kw = enkf_node_internalize(enkf_node , report_step);
+
+          if (internalize_kw) {
+            if (enkf_node_has_func(enkf_node , forward_load_func)) {
+              if (enkf_node_forward_load(enkf_node , load_context)) {
+                node_id_type node_id = {.report_step = report_step ,
+                                        .iens = iens };
+
+                enkf_node_store( enkf_node , result_fs, store_vectors , node_id );
+              } else {
+                forward_load_context_update_result(load_context, LOAD_FAILURE);
+                ert_log_add_fmt_message( 1 , NULL , "[%03d:%04d] Failed load data for FIELD node:%s.",iens , report_step , enkf_node_get_key( enkf_node ));
+
+                if (forward_load_context_accept_messages(load_context)) {
+                  char * msg = util_alloc_sprintf("Failed to load: %s at step:%d" , enkf_node_get_key( enkf_node ) , report_step);
+                  forward_load_context_add_message(load_context, msg);
+                  free( msg );
+                }
+              }
+            }
+          }
+        }
+      }
```

Comment:
I take it that this is mainly a copy-paste, but this would be much more readable if we went for early returns!

I count `if while if if if if/else ... if` nested levels, plus an extra level by the unnecessary block (`{...}`).

```c
if (! line 64) return;
if (! line 84) continue;
if (line 88) {
   ...
   continue;
}
if (line 91 && line 92) {
  ...
}
```
 ... would decrease indentation level by 5 steps, I believe.